catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/CountryCodeValidator.java

```diff
@@ -0,0 +1,107 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import com.google.common.base.Preconditions;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+
+/**
+ * Validates an attribute's value(s) against the ISO_3166-1 Alpha3 country codes.
+ * <p>
+ * Is capable of validating {@link String}s.
+ */
+public class CountryCodeValidator implements AttributeValidator {
+    private Set<String> countryCodes;
+
+    //TODO: Should there be an option to ignore case?
+    public CountryCodeValidator() {
+        setupISO3CountryCodes();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Validates only the values of {@code attribute} that are {@link String}s.
+     */
+    @Override
+    public Optional<AttributeValidationReport> validate(Attribute attribute) {
+        Preconditions.checkArgument(attribute != null, "The attribute cannot be null.");
+
+        final String attributeName = attribute.getName();
+
+        return attribute.getValues()
+                .stream()
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .filter(s -> !countryCodes.contains(s))
+                .map(s -> this.buildReport(attributeName, s))
+                .findFirst();
+    }
+
+    private AttributeValidationReport buildReport(String attributeName, String countryCodeValue) {
+        AttributeValidationReportImpl report = new AttributeValidationReportImpl();
+        report.addViolation(new ValidationViolationImpl(Collections.singleton(attributeName),
+                countryCodeValue + " is not a valid ISO_3166-1 Alpha3 country code.",
+                ValidationViolation.Severity.ERROR));
+
+        //TODO: Should the country codes be included in the error report?
+        countryCodes.forEach(report::addSuggestedValue);
+        return report;
+    }
+
+    private void setupISO3CountryCodes() {
+        countryCodes = Arrays.stream(Locale.getISOCountries())
+                .map(iso2CC -> new Locale("", iso2CC))
+                .map(Locale::getISO3Country)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
```

Comment:
I implemented them because other AttributeValidator implementations overrode them. I assumed the decision was made to do this for all validators for a particular reason. Perhaps to programmatically tell the difference between AttributeValidators without knowing the implementations.